app.js

```diff
@@ -48,10 +37,38 @@ angular.module('PathOfDamage', [])
         }
       }
     }
-    $scope.updateHits(true);
-    if (!skipStringify) {
-      $scope.stringifyUrlData();
+    if (!skipUpdates) {
+      $scope.updateHits();
+    }
+  };
+
+  $scope.updateHits = function (skipSerialization) {
+    for (var i = 0; i < $scope.hits.length - 1; i++) {
+      $scope.hits[i] = calcDamage($scope.hits[i].hit);
+    }
+    if (!skipSerialization) {
+      serializeData();
+    }
+  };
+
+  $scope.updateHitTable = function (index) {
```

Comment:
I couldn't think of anything better. It can modify the structure of the table regardless of which row you're editing, so updateHitRow or updateHit seemed misleading? I'm open to ideas =P